packages/agent/src/providers/TrustedEntitiesProvider.tsx

```diff
@@ -0,0 +1,69 @@
+import { OpenId4VcSiopHolderService } from '@credo-ts/openid4vc'
+import type { FullAppAgent } from 'packages/agent/src'
+import { type ReactNode, createContext, useContext, useEffect, useState } from 'react'
+
+export const TRUSTED_ENTITIES = [
+  'https://funke.animo.id/siop/0193687b-0c27-7b82-a686-ff857dc6bbb3',
+  'https://funke.animo.id/siop/0193687f-20d8-720a-9139-ed939ba510fa',
+] satisfies [string, ...string[]]
+
+type TrustedEntity = {
+  entity_id: string
+  organization_name: string
+  logo_uri?: string
+}
+
+type TrustedEntitiesContextType = {
+  trustedEntities: TrustedEntity[]
+  isLoading: boolean
+}
+
+const TrustedEntitiesContext = createContext<TrustedEntitiesContextType | undefined>(undefined)
+
+export function TrustedEntitiesProvider({ agent, children }: { agent: FullAppAgent; children: ReactNode }) {
+  const [isLoading, setIsLoading] = useState(true)
+  const [trustedEntities, setTrustedEntities] = useState<TrustedEntity[]>([])
+
+  useEffect(() => {
+    async function fetchTrustedEntities() {
+      try {
+        const openId4VcSiopHolderService = agent.dependencyManager.resolve(OpenId4VcSiopHolderService)
+
+        // Fetch configurations for all trusted entities
+        const entityPromises = TRUSTED_ENTITIES.map((entityId) =>
+          openId4VcSiopHolderService.fetchOpenIdFederationEntityConfiguration(agent.context, {
+            entityId,
+          })
+        )
```

Comment:
Why do'nt we just fetch an entity when we get a request? Why do we need to pre-fetch this?

Reply:
This was kinda because we always wanted to show all entities, but i guess it still doesn't hurt to just fetch them once on start up and use the cached verisons.